### PR TITLE
docs(dotnet): examples for dialog, download and filechooser

### PR DESCRIPTION
**Pull request by avodovnik**

### CODE REVIEW

**Review by pavelfeldman — APPROVED** [review]

File: docs/src/api/class-dialog.md [hunk 1]
```diff
@@ -80,6 +80,30 @@ with sync_playwright() as playwright:
...
+
+class DialogExample
+{
+    public static async Task Run()
+    {
+        using var playwright = await Playwright.CreateAsync();
+        var browser = await playwright.Chromium.LaunchAsync();
+        var page = await browser.NewPageAsync();
+
+        page.Dialog += async (_, dialog) =>
```
Comment (pavelfeldman) [inline docs/src/api/class-dialog.md:95]: Does C# support async event handlers?

File: docs/src/api/class-download.md [hunk 2]
```diff
@@ -48,6 +48,15 @@ download = download_info.value
 path = download.path()
 ```
 
+```csharp
+var downloadTask = page.WaitForDownloadAsync();
+await Task.WhenAll(downloadTask, page.ClickAsync("#downloadButton"))
+    .WithTimeout(1000)
```
Comment (pavelfeldman) [inline docs/src/api/class-download.md:54]: - There is a natural timeout on both waitForDownload and click, why do we add another one?
- I don't think we need ConfigureAwait(false) in any of our examples and code.

File: docs/src/api/class-dialog.md (hunk 1 above)
Comment (avodovnik) [reply docs/src/api/class-dialog.md:95]: Yup - like in the example. It makes calling `async` methods further easier :-).

File: docs/src/api/class-download.md [hunk 3]
```diff
@@ -48,6 +48,12 @@ download = download_info.value
 path = download.path()
 ```
 
+```csharp
+var downloadTask = page.WaitForDownloadAsync();
+await Task.WhenAll(downloadTask, page.ClickAsync("#downloadButton"));
+System.Console.WriteLine(await downloadTask.Result.PathAsync());
```
Comment (kblok) [inline docs/src/api/class-download.md:54]:
```suggestion
Console.WriteLine(await downloadTask.Result.PathAsync());
```

**Review by kblok — APPROVED** [review]

**Review by kblok — APPROVED** [review]

File: docs/src/api/class-dialog.md (hunk 1 above)
Comment (pavelfeldman) [reply docs/src/api/class-dialog.md:95]: We should follow up to fix this here (and maybe in other places).

**Pull request merged**